lib/3scale/backend/storage_async/client.rb

```diff
@@ -46,124 +48,13 @@ def initialize(opts)
           port ||= DEFAULT_PORT
 
           endpoint = Async::IO::Endpoint.tcp(host, port)
-          @redis_async = Async::Redis::Client.new(
+          @redis_async = Concurrent::ThreadLocalVar.new{ Async::Redis::Client.new(
```

Comment:
I'm ok with such an approach if using proper async is not trivial. But it appears to me that `async-redis` was built to handle concurrency. So mayeb gove a little more explanation why do we need a separate client per thread?

Reply:
`async-redis` relies on Fibers to implement concurrency, but concurrency doesn't mean parallelism. What `async-redis` does, AFAIK, is to switch between fibers on IO operations, e.g. when one fiber is waiting for answer from Redis. But all Fibers run in the same Thread. In Our case, we are calling `Storage.instance` from different Threads, `@redis_async` was being shared between threads and that caused an error `Fiber called across threads` which blocked the test.

@eguzki Does this make sense to you?